include/maidsafe/crux/detail/multiplexer.hpp

```diff
@@ -387,8 +389,9 @@ inline void multiplexer::do_start_receive()
 inline void multiplexer::discard_message() {
     boost::system::error_code error;
     endpoint_type remote_endpoint;
+    static char empty_buffer_storage;
     next_layer().receive_from
-        ( boost::asio::buffer(static_cast<char*>(nullptr), 0)
+        ( boost::asio::buffer(&empty_buffer_storage, 1)
```

Comment:
Should we hold of on merge for this @inetic or do you want to do a Jira task and add it as an update ?